### ISSUE TITLE
🐛 Bug Report: AWS S3 Publisher's `docsRouter` is consistently slow

### ISSUE THREAD

**Issue by chriskilding** [backstage/backstage#16852]
### 📜 Description

Our Backstage installation is instrumented with New Relic. This shows that the `docsRouter()` implementation in the AWS S3 Publisher (which serves Tech Docs from S3) is consistently quite slow. Specifically, I see that `GET /api/static/docs` requests typically take 1-1.2 seconds to complete (and can run to several seconds in some cases).

New Relic indicates that in a typical request to this endpoint, the breakdown of time spent looks roughly like this:

- Middleware: 70%
- External (calling S3 to fetch the Techdocs file): 25%
- The last couple of % comprise the DNS lookup, `http.Agent#createConnection`, and so on

Therefore the S3 call itself is **not** what takes the majority of the time. Instead, something in the S3 Publisher's `docsRouter()` logic - or perhaps some Node HTTP middleware involved - is taking the time. I'm not entirely sure what; that's one reason I have filed this issue.

The S3 Techdocs Publisher setup in the Backstage installation is vanilla and hasn't been wrapped with any custom handler code (that I know about), so I believe that the slowness I've seen is most likely down to how upstream Backstage code behaves.

### 👍 Expected behavior

When Techdocs are fetched from S3, the vast majority of the time taken should be in the nested S3 request. I.e. it should be I/O-bound on S3.

### 👎 Actual Behavior with Screenshots

When Techdocs are fetched from S3, a large amount of time is spent in what New Relic calls "middleware" (this could be the business logic around the GetObject request in `docsRouter()`, or it could be actual Node HTTP middleware libraries - I'm not sure).

In other words the request is not primarily I/O bound on S3, and appears to 'waste' a lot of time doing something else.

### 👟 Reproduction steps

1. Set up Backstage with the awsS3 Publisher configured to get Techdocs from S3.
2. Either load the Techdocs page (which should spawn a lot of requests to the relevant Techdocs endpoint) or make equivalent individual calls to the API.
3. See how long they take.

### 🖥️ Your Environment

Backstage v1.10.1

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None
